### ISSUE TITLE
Updating from 1.0.0-beta.14 to anything =>1.0.0-beta.15  broke my belongsTo relationship

### ISSUE THREAD

**Issue by jackmatt2** [emberjs/data#3063]
`model.save()` produces the following JSON for a new record (POST request.)  For a PUT request I have no issues.

1.0.0-beta.14 JSON
```json
{"code":"test","description":"test","partsUsed":false,"createUserId":null,"modifiedUserId":null,"version":null,"createDate":null,"modifiedDate":null,"category":"2","id":0}
```

1.0.0-beta.15 JSON
```json
{"code":"test","description":"test","partsUsed":false,"createUserId":null,"modifiedUserId":null,"version":null,"createDate":null,"modifiedDate":null,"category":null,"id":0}
```

Notice that the difference is that the category ID is null for beta.15.  When I click on the object from ember inspector and click on the category promise I can see the category updating as you would expect as I change the category via an Ember Select control.

boilerplate/models/mixins/audit-fields.js
```javascript
import Ember from 'ember';
import DS from 'ember-data';

export default Ember.Mixin.create({
    createUserId : DS.attr('string'),
    modifiedUserId : DS.attr('string'),
    version : DS.attr('number'),
    createDate  : DS.attr('date'),
    modifiedDate  : DS.attr('date'),

    displayName : function() {
        return this.toString();
    }.property()
});
```

app/models/category
```javascript
import DS from 'ember-data';
import AutoGen from './gen/category-gen';
import AuditFields from 'boilerplate/models/mixins/audit-fields';

export default DS.Model.extend(AutoGen, AuditFields, {
    displayName : function() {
        return this.get('description');
    }.property('description'),
});
```

app/models/gen/category-gen
```javascript
import Ember from 'ember';
import DS from 'ember-data';

export default Ember.Mixin.create({
    code : DS.attr('string'),
    description : DS.attr('string'),


    handlingCodes : DS.hasMany('handling-code', { async: true, inverse: 'category' })
});
```

app/models/handling-code
```javascript
import DS from 'ember-data';
import AutoGen from './gen/handling-code-gen';
import AuditFields from 'boilerplate/models/mixins/audit-fields';

export default DS.Model.extend(AutoGen, AuditFields, {});
```

app/models/gen/handling-code-gen
```javascript
import Ember from 'ember';
import DS from 'ember-data';

export default Ember.Mixin.create({
    code : DS.attr('string'),
    description : DS.attr('string'),
    partsUsed : DS.attr('boolean'),

    category : DS.belongsTo('category', {async: true })

});
```

Any ideas what I am doing wrong?  Is there an api change I am not aware of?

**Comment by igorT** [comment]
Hi, it's a bit hard for me to help, could you narrow down the issue at all? Also happy to pair/screenshare if you catch me online at IRC.

**Comment by jackmatt2** [comment]
@igorT it may be related to [this](https://github.com/emberjs/data/issues/2823) issue with Ember.Select.

**Comment by jackmatt2** [comment]
@igorT bellow is how my select elements are defined (does anything about this look incorrect to you?)

```javascript
                {{view "select"
                content=control.collection
                optionValuePath="content.id"
                optionLabelPath='content.name'
                selection='model.employee.content'
                prompt="Please select"
                }}
```

**Comment by igorT** [comment]
does changing                 selection='model.employee.content'
to                 selection='model.employee'
make a difference?

**Comment by omghax** [comment]
I just spent way too long tracking down a similar issue, so I'll leave my solution here in case someone finds it helpful. I have an app with a bunch of forms, and acceptance tests that verify the payload sent to the (Pretender) server.

After upgrading to ember-cli 0.2.3, I started seeing failures where any values bound to an Ember.Select weren't being updated. After some debugging, I noticed that the record was being saved *before* Ember.Select had updated its value.

Turns out something changed in ember-qunit at some point, possibly causing `test` to introduce a new runloop? I'm not entirely sure, but changing this:

```javascript
import { module } from 'qunit';
import { test } from 'ember-qunit';
```

into this:

```javascript
import { module, test } from 'qunit';
```

fixed the issue for me.
